### ISSUE TITLE
review all suppressed IntellijIdea inspections with comment 'till ...'

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#4681]
https://github.com/checkstyle/checkstyle/blob/master/config/intellij-idea-inspections.xml

comments like:
`Disabled till GSoC completion`
`disabled till https://githu......`
...
any other place where `... enabled="false" ...`

Sub issues: #3435 , #3301

TODO:
1) Active all suppressed by "till" inspections, share report with us to review and receive guidance on what to fix and how.
2) if no reason for suppression is provided , reason comment should be created (discussed with me).
3) all suppressed cases need to reviewed and fixed is reasonable (DO NOT blindly follow recommendations from inspection)

**Comment by Nimfadora** [comment]
@romani 
All suppressions with #till comment you can find [here](https://nimfadora.github.io/idea/suppressed.html)
List of suppressions without comments is [here](https://nimfadora.github.io/idea/without-comments.html), report on them is too big to be extracted, so if it is needed I will generate it on specific checks.

**Comment by romani** [comment]
@Nimfadora ,

> https://nimfadora.github.io/idea/suppressed.html

All link to sources are like 
`file:///home/valeria/IdeaProjects/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java#3422` , hard to use report from cell phone.

it would be awesome if you did search/replace for all links in files to github repository. `sed` could do this for you by one command. Please do. Looks like it not the last time you need to share violations.

**Comment by romani** [comment]
@Nimfadora , look at #4698, rebase after merge, it address the most huge in amount violations

**Comment by Nimfadora** [comment]
@romani [here](https://nimfadora.github.io/idea/suppressed_first.html) is the report on first part of undocumented violations 
StringEquals
StringEqualsIgnoreCase
StringTokenizer
SuperTearDownInFinally
SuspiciousLiteralUnderscore
SuspiciousLocalesLanguages
SwitchStatement
SwitchStatementWithTooManyBranches
SynchronizationOnStaticField
SyntaxError
SystemExit
SystemOutErr
SystemProperties
TailRecursion
TestMethodWithoutAssertion
ThisEscapedInConstructor
ThreeNegationsPerMethod
ThrowCaughtLocally
TodoComment
TooBroadCatch
TrailingSpacesInProperty
TransientFieldNotInitialized
TryFinallyCanBeTryWithResources
TsLint
TypeMayBeWeakened
UncheckedExceptionClass
UnclearBinaryExpression
UnnecessaryDefault
UnnecessaryUnicodeEscape
UnusedProperty
UnusedReturnValue
UseOfAnotherObjectsPrivateField
UseOfPropertiesAsHashtable
UtilityClass
UtilityClassCanBeEnum
VarargParameter
VariableNotUsedInsideIf
VtlReferencesInspection
WeakerAccess
WrongPropertyKeyValueDelimiter
XmlHighlighting
XmlPathReference
XmlUnboundNsPrefix
unused

**Comment by Nimfadora** [comment]
@romani I also add links to the "till ..." report, so you can review it in more comfortable way

**Comment by Nimfadora** [comment]
@romani UnnecessaryFinalOnLocalVariableOrParameter has 7904 violations, should I send separate report on this violation?

**Comment by romani** [comment]
>UnnecessaryFinalOnLocalVariableOrParameter has 7904 violations, should I send separate report on this violation?

Suppress it as it conflicting with our FinalVariableCheck

**Comment by rnveach** [comment]
> UnnecessaryFinalOnLocalVariableOrParameter 

Is there anyway to make it only do parameters?

**Comment by Nimfadora** [comment]
@romani UnqualifiedMethodAccess has 11547 violations what should I do with it?

**Comment by Nimfadora** [comment]
@rnveach yes we can

**Comment by rnveach** [comment]
> > Is there anyway to make it only do parameters and ignore variables?

> yes we can

@Nimfadora @romani We should enable it this way then as we wanted to do something similar in sevntu. See https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/581 .
@Nimfadora Please give us the number of violations when only doing parameters.

**Comment by Nimfadora** [comment]
@rnveach we have 314 errors when ignoring variables. [here is report](https://nimfadora.github.io/idea/suppressed_unnecessary_final.html)

**Comment by romani** [comment]
@Nimfadora ,

from https://nimfadora.github.io/idea/suppressed_first.html

>Unbound XML namespace prefix inspection   14 warnings 
XML highlighting inspection   127 errors 

Suppress by "there is not xsd or DTD for ant files https://ant.apache.org/faq.html#dtd, some other config files  also do not have schemes too"

**Comment by romani** [comment]
>Use of System.out or System.err inspection   9 error

 Suppress by javadoc tag in Main class.

**Comment by romani** [comment]
I experience extra violations(redundant suppression) in my IntelliJ 2017.1.4 in comparison TC (2017.1.2). Fix of them cause failure of TC, so IDEA version should be the same. 
I did not find way to download the same version as TC, filed request https://teamcity-support.jetbrains.com/hc/en-us/requests/1036431

**Comment by romani** [comment]
@Nimfadora , FYI

TC support replied to support request and uploaded to TC latest version of IDEA (2017.1.5) and I activated it in CI process, if you still experience different results of inspection validation it is recommended to upgrade IDEA to 2017.1.5.

**Comment by romani** [comment]
@Nimfadora , not all disabled inspections are commented.
Each rule has to have comment.

```
$ grep -B1 "enabled=\"false\" level" intellij-idea-inspections.xml | grep "enabled=\"false\"" | wc -l
145
$ grep -B1 "enabled=\"false\" level" intellij-idea-inspections.xml | grep "\-\->" | wc -l
141
```

there should be no "till GSoC" comments or similar.

**Comment by romani** [comment]
All works is done.
We end-up with 137 disabled inspections.
```
$ grep -B1 "enabled=\"false\" level" config/intellij-idea-inspections.xml | grep "enabled=\"false\"" | wc -l
137
$ grep -B1 "enabled=\"false\" level" config/intellij-idea-inspections.xml | grep "\-\->" | wc -l
137
```
